spec/helpers/lib/resourceHints.js

```diff
@@ -0,0 +1,151 @@
+const _ = require('lodash');
+const Code = require('code'),
+    expect = Code.expect;
+const Sinon = require('sinon');
+const Lab = require('lab'),
+    lab = exports.lab = Lab.script(),
+    describe = lab.experiment,
+    it = lab.it,
+    beforeEach = lab.beforeEach,
+    afterEach = lab.afterEach;
+const {
+    addResourceHint,
+    defaultResourceHintSate,
+    resourceHintAllowedTypes
+} = require('../../../helpers/lib/resourceHints');
+
+describe('resource hints', function () {
+
+    describe('addResourceHint', function () {
+
+        let sandbox;
+
+        beforeEach(done => {
+            sandbox = Sinon.createSandbox();
+            done();
+        });
+
+        afterEach(done => {
+            sandbox.restore();
+            done();
+        });
+
+        it("does work as expected with valid params", (done) => {
+
+            const globals = {resourceHints: []};
+            sandbox.spy(globals.resourceHints, 'push');
+
+            const src = '/my/styles.css';
+            const type = 'style';
+            const state = 'preload';
+            const expected = {src, state, type};
+
+            addResourceHint(globals, src, state, type);
+
+            expect(globals.resourceHints.push.called).to.equals(true);
+            expect(globals.resourceHints).to.have.length(1);
+            const result = _.head(globals.resourceHints);
+            expect(result).to.equals(expected);
+
+            done();
+        });
+
+        it('does not add a hint when provided path is invalid', (done) => {
+            const globals = {resourceHints: []};
+            sandbox.spy(globals.resourceHints, 'push');
+
+            const throws = [undefined, null, ''].map(s => {
+                const f = () => {
+                    addResourceHint(
+                        global,
+                        s,
+                        defaultResourceHintSate,
+                        resourceHintAllowedTypes.resourceHintStyleType
+                    );
+                }
+                return f;
+            });
+
+            throws.forEach(t => {
+                expect(t).to.throw();
+            });
```

Comment:
I think it can work like this
```suggestion
            [undefined, null, ''].forEach(input => {
                expect(
                    addResourceHint(
                        global,
                        input,
                        defaultResourceHintSate,
                        resourceHintAllowedTypes.resourceHintStyleType
                    );
                ).to.throw();
            });
```